code/modules/projectiles/guns/rifles.dm

```diff
@@ -1279,6 +1279,67 @@
 /obj/item/weapon/gun/rifle/l42a/training
 	current_mag = /obj/item/ammo_magazine/rifle/l42a/rubber
 
+//-------------------------------------------------------
+//-------------------------------------------------------
+
+//  M4RA Designated Marksmen Rifle
+
+/obj/item/weapon/gun/rifle/m4ra
+	name = "\improper M4RA Designated Marksman Rifle"
+	desc = "The M4RA is a designated marksman rifle fielded by the USCM. It was recently caught in a fierce competition with the L4 Battle Rifle, but ultimately the Colonial Marine Corps deemed the L4 too expensive to adopt and continued using the M4RA. Featuring bullpup ergonomics and utilising a higher velocity round than the M41, this is the perfect pick for any marine who prefers to keep a distance."
+	icon_state = "m4ra"
+	item_state = "m4ra" //PLACEHOLDER
+	fire_sound = 'sound/weapons/gun_m4ra.ogg'
+	current_mag = /obj/item/ammo_magazine/rifle/m4radmr
+	force = 16
+	attachable_allowed = list(
+						/obj/item/attachable/suppressor,
+						/obj/item/attachable/extended_barrel,
+						/obj/item/attachable/heavy_barrel,
+						/obj/item/attachable/verticalgrip,
+						/obj/item/attachable/angledgrip,
+						/obj/item/attachable/flashlight/grip,
+						/obj/item/attachable/bipod,
+						/obj/item/attachable/compensator,
+						/obj/item/attachable/attached_gun/grenade,
+						/obj/item/attachable/attached_gun/flamer,
+						/obj/item/attachable/attached_gun/shotgun,
+						/obj/item/attachable/attached_gun/extinguisher,
+						/obj/item/attachable/scope,
+						/obj/item/attachable/scope/mini,
+						/obj/item/attachable/reddot,
+						/obj/item/attachable/reflex,
+						/obj/item/attachable/magnetic_harness,
+						/obj/item/attachable/stock/m4ra
+						)
+
+	flags_gun_features = GUN_AUTO_EJECTOR|GUN_CAN_POINTBLANK|GUN_AMMO_COUNTER
+	wield_delay = WIELD_DELAY_SLOW
+
+	flags_item = TWOHANDED
```

Comment:
isn't this inherited

Reply:
yep. not needed here, inherent to rfiles